R/layout.R

```diff
@@ -123,13 +184,14 @@ Layout <- ggproto("Layout", NULL,
     )
   },
 
-  train_ranges = function(self, coord) {
+  train_ranges = function(self) {
     compute_range <- function(ix, iy) {
       # TODO: change coord_train method to take individual x and y scales
-      coord$train(list(x = self$panel_scales$x[[ix]], y = self$panel_scales$y[[iy]]))
+      scales <- list(x = self$panel_scales$x[[ix]], y = self$panel_scales$y[[iy]])
+      self$coord$train(scales, self$coord_params)
     }
     # Switch position of all scales if CoordFlip
-    if (inherits(coord, "CoordFlip") || (inherits(coord, "CoordPolar") && coord$theta == "y")) {
+    if (inherits(self$coord, "CoordFlip") || (inherits(self$coord, "CoordPolar") && self$coord$theta == "y")) {
```

Comment:
I ended up calling the method `modify_scales()` since it does in-place modification of the scales. I'm not 100% sure this is correct so I'm not planning on documenting as an extension point.